src/tools/run-make-support/src/external_deps/cargo.rs

```diff
@@ -1,11 +1,17 @@
 use crate::command::Command;
-use crate::env_var;
 use crate::util::set_host_compiler_dylib_path;
 
-/// Returns a command that can be used to invoke cargo. The cargo is provided by compiletest
-/// through the `CARGO` env var.
+/// Returns a command that can be used to invoke in-tree cargo. The cargo is provided by compiletest
+/// through the `CARGO` env var, and is **only** available for the `run-make-cargo` test suite.
 pub fn cargo() -> Command {
-    let mut cmd = Command::new(env_var("CARGO"));
+    let cargo_path = std::env::var("CARGO").unwrap_or_else(|e| {
```

Comment:
Remark: in theory, I could gate `run_make_support::cargo` behind a cargo feature flag (`#[cfg(feature = "is_run_make_cargo_suite")]`) or sth, but that feels like overkill, and makes the `run_make_support` build logic more complicated.